src/components/Tooltip/index.tsx

```diff
@@ -15,14 +19,22 @@ interface TooltipProps extends Omit<PopoverProps, 'content'> {
 
 interface TooltipContentProps extends Omit<PopoverProps, 'content'> {
   content: ReactNode
+  container?: JSX.Element
 }
 
 export default function Tooltip({ text, ...rest }: TooltipProps) {
   return <Popover content={<TooltipContainer>{text}</TooltipContainer>} {...rest} />
 }
 
-function TooltipContent({ content, ...rest }: TooltipContentProps) {
-  return <Popover content={<TooltipContainer>{content}</TooltipContainer>} {...rest} />
+function TooltipContent({ content, container, ...rest }: TooltipContentProps) {
+  return (
+    <Popover
+      content={
+        container ? cloneElement(container, { children: content }) : <TooltipContainer>{content}</TooltipContainer>
```

Comment:
clone element smells, almost never a good idea since you're overriding one of the props (in this case the children of the component). just render the Popover in the one case where you don't want a tooltip container

Reply:
> just render the Popover in the one case where you don't want a tooltip container
not quite sure what you mean by that

I changed container to a functional component.